code/modules/mechs/mech_damage.dm

```diff
@@ -40,36 +40,69 @@
 	return def_zone //Careful with effects, mechs shouldn't be stunned
 
 /mob/living/exosuit/getarmor(def_zone, type)
-	. = ..()
 	if(body?.armor_plate)
 		var/body_armor = body.armor_plate?.armor.getRating(type)
-		if(body_armor) . += body_armor
+		if(body_armor) return body_armor
+	return 0
 
 /mob/living/exosuit/updatehealth()
 	if(body) maxHealth = body.mech_health
 	health = maxHealth - (getFireLoss() + getBruteLoss())
 
-/mob/living/exosuit/adjustFireLoss(amount, obj/item/mech_component/MC = pick(list(arms, legs, body, head)))
-	if(MC)
-		MC.take_burn_damage(amount)
-		MC.update_health()
+/mob/living/exosuit/adjustFireLoss(amount, obj/item/mech_component/MC = null)
+	if(!MC)
+		MC = pick(list(arms, legs, body, head))
+	MC.take_burn_damage(amount)
+	MC.update_health()
 
-/mob/living/exosuit/adjustBruteLoss(amount, obj/item/mech_component/MC = pick(list(arms, legs, body, head)))
-	if(MC)
-		MC.take_brute_damage(amount)
-		MC.update_health()
+/mob/living/exosuit/adjustBruteLoss(amount, obj/item/mech_component/MC = null)
+	if(!MC)
+		MC = pick(list(arms, legs, body, head))
+	MC.take_brute_damage(amount)
+	MC.update_health()
 
 /mob/living/exosuit/proc/zoneToComponent(zone)
 	switch(zone)
-		if(BP_EYES, BP_HEAD) return head
+		if(BP_EYES, BP_HEAD, BP_MOUTH) return head
 		if(BP_L_ARM, BP_R_ARM) return arms
-		if(BP_L_LEG, BP_R_LEG) return legs
+		if(BP_L_LEG, BP_R_LEG, BP_GROIN) return legs
 		else return body
 
 
 /mob/living/exosuit/apply_damage(damage = 0, damagetype = BRUTE, def_zone = null, sharp = FALSE, edge = FALSE, obj/used_weapon = null)
-	. = ..()
+	switch(damagetype)
+		if(BRUTE)
+			adjustBruteLoss(damage, def_zone)
+			return TRUE
+		if(BURN)
+			adjustFireLoss(damage, def_zone)
+			return TRUE
 	updatehealth()
```

Comment:
Do we really need to updatehealth if no damage is taken?

Reply:
meh , it was just there when i copy pasted the proc and adapted it , if i wanted to do performance stuff i would go more indepth.